pandas/io/html.py

```diff
@@ -741,7 +741,7 @@ def _parse(flavor, io, match, header, index_col, skiprows,
                                       parse_dates=parse_dates,
                                       tupleize_cols=tupleize_cols,
                                       thousands=thousands))
-        except StopIteration:  # empty table
```

Comment:
Telling me it's not a very good idea is not very helpful feedback given that that was what was written before in the first place.  First, why is it not a good idea?  And second, what might you suggest as alternatives to doing a `try-except` block?  Direct checks of whether the table is empty come to mind, but if there are others, it would be good to know